modules/phase_field/include/utils/MisorientationAngleCalculator.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "Moose.h"
+#include "EulerAngles.h"
+#include "EulerAngleProvider.h"
+
+typedef Eigen::Quaternion<Real> quatReal;
+struct misoriAngle_isTwining{Real misor; bool isTwinning; std::string twinType;};
+
+/**
+ * This is an orientation difference calculator, inspired by MTEX.
+ * However, it is an incomplete version and needs further optimization TODO.
+ */
+class MisorientationAngleCalculator
```

Comment:
You might consider making this a namespace instead. There are no instance variables nor instance methods. It's just a list of static methods.

Reply:
already edited